extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/MongoUtilities.java

```diff
@@ -94,4 +99,26 @@ static List<Bson> partitionAggregate(int totalParallelism, int processorIndex, b
     public static BsonTimestamp bsonTimestampFromTimeMillis(long time) {
         return new BsonTimestamp((int) MILLISECONDS.toSeconds(time), 0);
     }
+    /**
+     * Converts given bson timestamp to unix epoch.
+     */
+    @Nullable
+    public static LocalDateTime bsonDateTimeToLocalDateTime(@Nullable BsonDateTime time) {
+        if (time == null) {
+            return null;
+        }
+        Instant instant = Instant.ofEpochMilli(time.getValue());
```

Comment:
You can use 
`LocalDateTime ldt = LocalDateTime.ofEpochSecond(seconds, nanoSeconds, ZoneOffset.UTC);` 
if you want it . It will save one extra object creation , the Instant object

Reply:
https://github.com/hazelcast/hazelcast/pull/23959/commits/dde026332ddc63851a52b3c36d5ad9801c6366b9